CONFIGURE.md

```diff
@@ -560,3 +633,92 @@ To notify the agent which extensions it should load see [Loading extensions](Ext
+- rollingappender - logs written to this log sink are written to a file with a rolling policy
+- stdout - logs written to this log sink are written to the standard output
+- stderr - logs written to this log sink are written to the standard error
+- syslog - logs written to this log sink are written to the syslog
+- alert - logs written to this log sink are critical logs forwarded through HTTP as alerts
+- nullappender - logs written to this log sink are discarded
+
+Defining a new appender can be done in the format `appender.<appender name>=<appender type>` followed by defining its properties. For example here is how to define a rolling file appender:
+
+    # in minifi-log.properties
+    appender.rolling=rollingappender
+    appender.rolling.directory=${MINIFI_HOME}/logs
+    appender.rolling.file_name=minifi-app.log
+    appender.rolling.max_files=3
+    appender.rolling.max_file_size=5 MB
+
+Here's an example of an alert appender with its available properties:
+
+    # in minifi-log.properties
+    appender.alert1=alert
+    appender.alert1.url=<URL>
+    appender.alert1.filter=<regex pattern to match logs against>
+    appender.alert1.rate.limit=10 min
+    appender.alert1.flush.period=5 s
+    appender.alert1.batch.size=100 KB
+    appender.alert1.buffer.limit=1 MB
+    appender.alert1.level=TRACE
+    appender.alert1.ssl.context.service=<Name of the SSLContextService>
+
+### Log levels
+After the appenders are defined you can set the log level and logging target for each of them. Appenders can be set to log everything on a specific log level, but you can also define some appenders to only write logs from specific namespaces or classes with different log levels. The log level can be set to one of the following values: OFF, TRACE, DEBUG, INFO, WARN, ERROR. The log levels can be set in the following way:
+
+    # in minifi-log.properties
+    # Set the rolling appender to log everything with INFO level
+    logger.root=INFO,rolling
+    # But only write logs from the org::apache::nifi::minifi::FlowController class with ERROR level
+    logger.org::apache::nifi::minifi::core::logging::FlowController=ERROR,rolling
+
+    # Write all error logs in the minifi namespace to the alert1 appender
+    logger.org::apache::nifi::minifi=ERROR,alert1
+
+    # Logs from the FlowController class should be written to the stderr and rolling appenders with DEBUG level
+    logger.org::apache::nifi::minifi::FlowController=DEBUG,stderr,rolling
+
+    # Log warnings and errors from the LoggerConfiguration class to all appenders
+    logger.org::apache::nifi::minifi::core::logging::LoggerConfiguration=WARN
+
+### Log pattern
+The log pattern can be changed if needed using the `spdlog.pattern` property. The default log pattern is the following, but there are additional examples in the default minifi-log.properties file:
+
+    # in minifi-log.properties
+    spdlog.pattern=[%Y-%m-%d %H:%M:%S.%e] [%n] [%l] %v
```

Comment:
we could link to https://github.com/gabime/spdlog/wiki/3.-Custom-formatting#customizing-format-using-set_pattern, as the meaning of some of the pattern placeholders is not obvious

Reply:
Added link in 4e4434732dac8f91122fc910ea067c7c752283c6